model_compression_toolkit/core/common/graph/base_node.py

```diff
@@ -445,15 +515,24 @@ def get_unique_activation_candidates(self) -> List[Any]:
                              and not seen_candidates.add(candidate.activation_quantization_cfg)]
         return unique_candidates
 
-    def has_weights_quantization_enabled_candidate(self) -> bool:
+    def has_kernel_quantization_enabled_candidate(self, fw_info) -> bool:
         """
-        Checks whether the node has quantization configuration candidates that enable weights quantization.
+        Checks whether the node has quantization configuration candidates that enable kenrel attribute weights quantization.
+
+        Args:
+            fw_info: Framework info to decide which attributes should be quantized.
 
         Returns: True if the node has at list one quantization configuration candidate with weights quantization enabled.
         """
+        # This method is only used for Mixed precision purposes
+        kernel_attr = fw_info.get_kernel_op_attributes(self.type)
+        if kernel_attr[0] is not None:
+            kernel_attr = kernel_attr[0]
+        else:
+            return False
```

Comment:
This function is a bit confusing since it can return a boolean or the kernel string... Why not return None? (I know it's not ideal as well, but I think it's a bit better)

Reply:
Removing the function, we are only using it in one place, so I just implement the validation there, it shouldn't be part of BaseNode